lib/optimizely/event/batch_event_processor.rb

```diff
@@ -37,15 +37,15 @@ class BatchEventProcessor < EventProcessor
 
     def initialize(
       event_queue: SizedQueue.new(DEFAULT_QUEUE_CAPACITY),
-      event_dispatcher: Optimizely::EventDispatcher.new,
+      event_dispatcher: nil,
       batch_size: DEFAULT_BATCH_SIZE,
       flush_interval: DEFAULT_BATCH_INTERVAL,
       logger: NoOpLogger.new,
       notification_center: nil
     )
       @event_queue = event_queue
       @logger = logger
-      @event_dispatcher = event_dispatcher
+      @event_dispatcher = event_dispatcher || EventDispatcher.new(logger: @logger)
```

Comment:
keyword args can be used within the method signature directly, there's no need to do all the manual defaulting within the body.

e.g.
```ruby
def initialize(
  logger: NoOpLogger.new,
  event_dispatcher: Optimizely::EventDispatcher.new(logger: logger)
)
```